default.nix

```diff
@@ -9,48 +9,80 @@ let
     inherit system;
   };
 
-  nix-dev = pkgs.stdenv.mkDerivation {
-    name = "nix-dev";
-    src = ./.;
-    nativeBuildInputs = with pkgs.python310.pkgs; [
-      linkify-it-py
-      myst-parser
-      sphinx
-      sphinx-book-theme
-      sphinx-copybutton
-      sphinx-design
-      sphinx-notfound-page
-      sphinx-sitemap
-    ];
-    buildPhase = ''
-      make html
-    '';
-    installPhase =
-      let
-        # Various versions of the Nix manuals, grep for (nix-manual)= to find where they are displayed
-        # FIXME: This requires human interaction to update! See ./CONTRIBUTING.md for details.
-        releases = [
-          "2.19"
-          "2.18"
-          "2.13"
-        ];
-        inputName = version: pkgs.lib.strings.replaceStrings [ "." ] [ "-" ] version;
-        src = version: inputs."nix_${inputName version}";
-        manual = version: (import (src version)).default.doc;
-        copy = version: ''
-          cp -Rf ${manual version}/share/doc/nix/manual/* $out/manual/nix/${version}
+  nix-dev =
+    let
+      # Various versions of the Nix manuals, grep for (nix-manual)= to find where they are displayed
+      # FIXME: This requires human interaction to update! See ./CONTRIBUTING.md for details.
+      nix-releases = {
+        latest = "2.19";
+        rolling = "2.18";
+        stable = "2.18";
+        prev-stable = "2.13";
+      };
+      nixpkgs-releases = {
+        stable = "23.11";
+        prev-stable = "23.05";
+      };
```

Comment:
These here could be avoided by calling the `niv` entries `nix-latest`, `nix-rolling`, ..., `nixpkgs-stable`, `nixpkgs-prev-stable` instead. That way we never need to manually edit a file to update the versions, but can just do

```
niv update nix-latest --branch release-2.20`
```

Reply:
Reworked it. It's a bit less of a pain now, but still quite a pain.